Library/Homebrew/test/Rakefile

```diff
@@ -1,8 +1,14 @@
 require "rake"
 require "rake/testtask"
 
+def mac?
+  /darwin/i === RUBY_PLATFORM && !ENV["HOMEBREW_TEST_GENERIC_OS"]
+end
+
 TEST_DIRECTORY = File.dirname(File.expand_path(__FILE__))
-TEST_FILES = Dir["#{TEST_DIRECTORY}/test_*.rb"]
+TEST_FILES = Dir["#{TEST_DIRECTORY}/test_*.rb"].reject do |f|
+  f.include?("/test_os_mac_") && !mac?
+end
```

Comment:
That's a convincing argument. I hope others will also join the discussion, so it's not only the two of us exchanging opinions. Maybe there are options we haven't thought of yet.